### ISSUE TITLE
Training TinyViT on custom dataset

### ISSUE THREAD

**Issue by abhishekmann** [microsoft/Cream#144]
Can you please share a recipe on transfer learning of TinyViT pretrained on Imagenet 1K/22K to a different dataset including knowledge distillation from a teacher for the same downstream task. Thanks.

**Comment by wkcn** [comment]
Hi @abhishekmann, thanks for your attention to our work!

The hyper-parameter can refer to [moco-v3 for linear probe](https://github.com/facebookresearch/moco-v3), [cdfsl-benchmark for few-shot learning](https://github.com/IBM/cdfsl-benchmark), [mmdetection for object detection](https://github.com/open-mmlab/mmdetection).

For transfer learning on image classification, you can refer to our another paper: [MiniViT, Section 4.4 Transfer Learning Results](https://openaccess.thecvf.com/content/CVPR2022/papers/Zhang_MiniViT_Compressing_Vision_Transformers_With_Weight_Multiplexing_CVPR_2022_paper.pdf).

We only apply knowledge distillation to pretrain TinyViT on ImageNet 22k.

If you want to apply knowledge distillation on downstrem datasets, the teacher model should be fine-tuned firstly, then distill the student model by the same hyper-parameter except that the drop path rate is set to 0. [Saving Teacher logits](https://github.com/microsoft/Cream/blob/main/TinyViT/docs/SAVE_TEACHER_LOGITS.md) is available.

**Comment by Shyamji07** [comment]
Hi is there any code available for finetuning the model on custom classes and dataset?
Thanks :)

**Comment by wkcn** [comment]
@Shyamji07 Thanks for your attention to our work!

Sorry that we did not provide the finetuning script on custom dataset. 
You can refer to our [hyper-parameter](https://github.com/microsoft/Cream/blob/main/TinyViT/configs/22kto1k/tiny_vit_21m_22kto1k.yaml) and the setting in [MiniViT Sec 4.4](https://openaccess.thecvf.com/content/CVPR2022/papers/Zhang_MiniViT_Compressing_Vision_Transformers_With_Weight_Multiplexing_CVPR_2022_paper.pdf) for downstream classification datasets.

**Comment by Shyamji07** [comment]
> @Shyamji07 Thanks for your attention to our work!
> 
> Sorry that we did not provide the finetuning script on custom dataset. You can refer to our [hyper-parameter](https://github.com/microsoft/Cream/blob/main/TinyViT/configs/22kto1k/tiny_vit_21m_22kto1k.yaml) and the setting in [MiniViT Sec 4.4](https://openaccess.thecvf.com/content/CVPR2022/papers/Zhang_MiniViT_Compressing_Vision_Transformers_With_Weight_Multiplexing_CVPR_2022_paper.pdf) for downstream classification datasets.

Thank you for the response @wkcn 
Will follow these resources.

**Comment by wkcn** [comment]
@Shyamji07 TinyViT has been integrated into [mmpretrain](https://github.com/open-mmlab/mmpretrain) and [timm](https://github.com/huggingface/pytorch-image-models). You can refer to their fine-tune scripts : )

**Comment by Shyamji07** [comment]
> @Shyamji07 TinyViT has been integrated into [mmpretrain](https://github.com/open-mmlab/mmpretrain) and [timm](https://github.com/huggingface/pytorch-image-models). You can refer to their fine-tune scripts : )

Thanks a ton @wkcn :)
